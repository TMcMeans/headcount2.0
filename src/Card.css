.card {
  border: 2px solid black;
  color: black;
  font-size: 16px;
  flex: 1 0 200px;
  box-sizing: border-box;
  margin: 1rem .25em;
}

.red {
  color: red;
}

.green {
  color: green;
}


@media screen and (min-width: 40em) {
    .card {
       max-width: calc(50% -  1em);
    }
}
 
@media screen and (min-width: 60em) {
    .card {
        max-width: calc(25% - 1em);
    }
}

/* Add styling to cards */