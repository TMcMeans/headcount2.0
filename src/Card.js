import React from 'react'
import PropTypes from 'prop-types'


import './Card.css'

const Card = ({district}) => {
  const dataPoints = Object.keys(district.data)

  return(
    <div className="card">
      <h3 className="school-name">{district.school}</h3>
      <ul className="school-data">
        {dataPoints.map( year => <li className={district.data[year] < 0.5  ? 'red' : 'green'}>{year} : {district.data[year]} </li>)}
      </ul>
    </div>
  )
}

Card.propTypes = {
  district: PropTypes.shape({
     location: PropTypes.string,
     data: PropTypes.object
  }),
}

export default Card; 