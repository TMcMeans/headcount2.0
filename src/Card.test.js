import React from 'react';
import { shallow, mount } from 'enzyme';

import Card from './Card.js'
import DistrictRepository from './helper.js';
import kinderData from './data/kindergartners_in_full_day_program.js';

describe('Card', () => {
  const district = {school: 'Colorado', data: {2016: 0.33677}}

  it('should match snapshot', () => {
    const wrapper = shallow(<Card district={district} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have a class of card', () => {
    const wrapper = shallow(<Card district={district} />)
    expect(wrapper.exists('.card')).toBe(true)
  })

  it('should render card data filtered by color', () => {
    const district = {school: 'Colorado', data: {2016: 0.33677}}
    const wrapper = shallow(<Card district={district} />)

    expect(wrapper.find('li').hasClass('red'))
  })
})


