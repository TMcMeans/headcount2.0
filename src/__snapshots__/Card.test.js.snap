// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div
  className="card"
  onClick={[Function]}
>
  <h3
    className="school-name"
  >
    Colorado
  </h3>
  <ul
    className="school-data"
  >
    <li
      className="red"
      key="1538860756562"
    >
      2016
       : 
      0.33677
       
    </li>
  </ul>
</div>
`;
